inginious/agent/docker_agent/__init__.py

```diff
@@ -441,15 +440,27 @@ async def create_student_container(self, parent_info, socket_id, environment_nam
                 return
 
             environment = self._containers[environment_type][environment_name]["id"]
-            runtime = self._containers[environment_type][environment_name]["runtime"]
+
+            # If the user did not enter any environment name as input, it is "default"
+            if environment_name != "default":
+                runtime = environment_name
+            elif run_as_root:
+                runtime_name = {k for k in self._runtimes if self._runtimes[k].run_as_root}.pop()
+                runtime = self._runtimes[runtime_name].runtime
+            else:
+                runtime_name = {k for k in self._runtimes if not self._runtimes[k].run_as_root}.pop()
+                runtime = self._runtimes[runtime_name].runtime
 
             ports_needed = [22] if ssh else []
             ports = {}
             for p in ports_needed:
-                if len(self._external_ports) == 0:
+                if self._external_ports is None or len(self._external_ports) == 0:
```

Comment:
``if not self._external_ports`` should probably also work for both conditions

Reply:
Indeed :)